src/main/java/seedu/address/logic/commands/EditLessonCommand.java

```diff
@@ -0,0 +1,255 @@
+        }
+
+        Lesson lessonToEdit = targetModuleClass.getLessons().get(lessonIndex.getZeroBased());
+        Lesson editedLesson = createEditedLesson(lessonToEdit, editLessonDescriptor);
+        ModuleClass modifiedModuleClass = createModifiedModuleClass(targetModuleClass, lessonToEdit, editedLesson);
+
+        model.setModuleClass(targetModuleClass, modifiedModuleClass);
+        model.updateFilteredModuleClassList(PREDICATE_SHOW_ALL_MODULE_CLASS);
+        String message = String.format(MESSAGE_EDIT_LESSON_SUCCESS, editedLesson);
+        model.commit(message);
+        return new CommandResult(message);
+    }
+
+    /**
+     * Creates and returns a {@code Lesson} with the details of {@code lessonToEdit}
+     * edited with {@code editLessonDescriptor}.
+     */
+    private static Lesson createEditedLesson(
+            Lesson lessonToEdit, EditLessonDescriptor editLessonDescriptor) {
+        assert lessonToEdit != null;
+        LocalTime updatedStartTime = editLessonDescriptor.getStartTime().orElse(lessonToEdit.getStartTime());
+        LocalTime updatedEndTime = editLessonDescriptor.getEndTime().orElse(lessonToEdit.getEndTime());
+        NumberOfOccurrences originalNumberOfOccurrences = lessonToEdit.getNumberOfOccurrences();
+        Day updatedDay = editLessonDescriptor.getDay().orElse(lessonToEdit.getDay());
+        Venue updatedVenue = editLessonDescriptor.getVenue().orElse(lessonToEdit.getVenue());
+
+        return new Lesson(updatedStartTime, updatedEndTime, updatedDay, originalNumberOfOccurrences, updatedVenue);
+    }
+
+    /**
+     * Adds all lessons in the target module class to the new module class, other than the lesson to edit.
+     * The {@code editedLesson} is added in place of the {@code lessonToEdit}.
+     *
+     * @throws CommandException if the {@code editedLesson} already exists.
+     */
+    private static ModuleClass createModifiedModuleClass(
+            ModuleClass targetModuleClass, Lesson lessonToEdit, Lesson editedLesson) throws CommandException {
+        assert targetModuleClass != null;
+        assert lessonToEdit != null;
+        assert editedLesson != null;
+
+        if (!lessonToEdit.isSameLesson(editedLesson) && targetModuleClass.hasLesson(editedLesson)) {
+            throw new CommandException(MESSAGE_DUPLICATE_LESSON);
+        }
+
+        Name moduleClassName = targetModuleClass.getName();
+        Set<UUID> studentsIds = new HashSet<>(targetModuleClass.getStudentUuids());
+        List<Lesson> listOfLessons = targetModuleClass.getLessons();
+        List<Lesson> editedListOfLessons = new ArrayList<>();
+        for (Lesson lesson: listOfLessons) {
+            if (lesson.equals(lessonToEdit)) {
+                editedListOfLessons.add(editedLesson);
+            } else {
+                editedListOfLessons.add(lesson);
+            }
+        }
+        assert listOfLessons.size() == editedListOfLessons.size();
```

Comment:
We can consider adding this for ```AddLessonCommand``` and ```DeleteLessonCommand``` as well

Reply:
Agreed

Reply:
👍🏻